cnd/src/storage/db/wrapper_types.rs

```diff
@@ -95,3 +105,26 @@ impl fmt::Display for Erc20Amount {
         write!(f, "{}", self.0.to_wei_dec())
     }
 }
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct WeiPerSat(asset::Erc20Quantity);
+
+impl FromStr for WeiPerSat {
+    type Err = crate::asset::ethereum::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        asset::Erc20Quantity::from_wei_dec_str(s).map(Self)
+    }
+}
+
+impl fmt::Display for WeiPerSat {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0.to_wei_dec())
+    }
+}
+
+impl From<Text<WeiPerSat>> for Price<asset::Bitcoin, Erc20Quantity> {
+    fn from(rate: Text<WeiPerSat>) -> Self {
```

Comment:
My current bug bear; can we not introduce multiple terms for the same thing please. We are creating a price from some input value, why use 'rate'? Perhaps 'price' or 'p' or even 'value' if you don't want to use price.